### ISSUE TITLE
Getting Duplicate Notifications (tested on Android)

### ISSUE THREAD

**Issue by momo1** [evollu/react-native-fcm#727]
Hi,

I'm getting duplicate notifications on Android (haven't tested ios). My code is quite simple since I'm just starting out:

```
export default class NotificationService {
    static initHandler() {
      FCM.on(FCMEvent.Notification, async (notification) => this._handleNotification(notification));
    }
    static _handleNotification(notification) {
      console.log("Got " + notification.messageId);
    }
}
```

and in App.js

```
NotificationService.initHandler();
export const App = StackNavigator({
  Login: {screen: Login}
});
```
When I push a message from the backend service I get the same alert on the phone multiple times. I also keep getting previous alerts. The backend is not the problem. I checked that the backend is sending each notification once and I have the same app implemented 100% in Kotlin which works perfectly. 

I'm fairly certain that I'm not registering multiple handlers since I checked that initHandler() is only called once. I'm very new to RN, am I doing something wrong or is there a known issue?

* versions:     "react-native-fcm": "^11.0.2" and   "react-native": "0.51.0",
* Device: Samsung S8, Android 7.0
* This happens no matter what the app state is, foreground, background or stopped (cleared from task list).

Thanks!

**Comment by evollu** [comment]
1. `I get the same alert on the phone multiple times`, you mean banner? If so, it has nothing to do with javascript, perhaps you registered the service in android multiple times. check that
2. what is your payload sent to firebase?
3. try local notification and see if shows duplicate banner

**Comment by momo1** [comment]
1. Sorry what I meant when I said I get the same notification is that the handler _handleNotification() invokes multiple times and so logs to the console multiple times - there's no banner since it's data only. I have no other firebase handlers on the phone for this project ID. I have only one version of the of the RN app on the phone and the other Kotlin implementation that I mentioned is completely uninstalled. Also as I mentioned I verified with a log statement and debugger that only one handler is being registered.

2. The firebase payload is very simple:

Request:
```
{
  "data":{ 
    "messageId":"6b31a..."
    },
    "to":"<firebaseToken>"
}
```
Response from Firebase:
```
{"multicast_id":"....", "success":1,"failure":0,"canonical_ids":0,"results":[{"message_id":"...."}]}
```

3. Again sorry for the confusion, there is no banner. It's data only but if you still think trying a local notification would help I can set it up.

Thanks again.

**Comment by evollu** [comment]
have you load the react app multiple times?
`NotificationService.initHandler();` might execute every time you do a app reload.

**Comment by momo1** [comment]
Yes I use the reload functionality quite a bit. I thought it might be the problem so I tried stopping the app uninstalling and reinstalling and not using RR. That didn't solve it. I also generated a standalone APK thinking maybe something in my IDE is interfering with it. Still didn't fix the problem.

**Comment by evollu** [comment]
does local notification have the issue?

**Comment by prasanthyejje** [comment]
I am Also getting Duplicate notifications..there is no problem from server side ..could you help me please @momo1 @evollu

**Comment by huynhtansi** [comment]
Hi @evollu , on Android when receive message you perform                    context.sendOrderedBroadcast(message, null) (Line 44 in MessagingService). So all receivers which register IntentFilter("com.evollu.react.fcm.ReceiveNotification") can listen this event.

In my case, if I create two applications which using react-native-fcm, they can receive messages each others.

**Comment by evollu** [comment]
@huynhtansi are you using the latest version? this issue has been fixed to use `LocalBroadcastManager`

**Comment by huynhtansi** [comment]
@evollu , thanks for pointing it out. I will update my repo.

**Comment by prasanthyejje** [comment]
--> I am Using RN 0.51 and "npm install react-native-fcm --save" 
--> Using Redux In My Project.
My Code Is: 

`   this.notificationListener = FCM.on(FCMEvent.Notification, async (notif) => {
                console.log("notificationListener");

                if (notif.isChatNotification == 'False') {
                        console.log(notif);
                    this.props.NewRequestNotificationReceived({ notif });

                    Alert.alert(
                        'Notification',
                        // notif.body,
                        notif.body,
                        [
                            { text: 'OK' }
                        ],
                        { cancelable: false }
                    )
                }`

But When App is in Foreground I am Getting Duplicate Notifications . Please Help Me to Solve this @evollu

**Comment by evollu** [comment]
@prasanthyejje what does NewRequestNotificationReceived do.

**Comment by prasanthyejje** [comment]
@evollu  I am Using Redux. when Notification Received, i am Passing the Notification Data To Reducer Component By Using 
> this.props.NewRequestNotificationReceived({ notif });

**Comment by evollu** [comment]
`When App is in Foreground I am Getting Duplicate Notifications` do you mean the callback gets called twice? Are the 2 notificaitons exactly the same?

**Comment by prasanthyejje** [comment]
@evollu  Getting Same Notification Multiple times(Duplicate Notifications),when I debug i found that, The FCM.ON executing Multiple Times.. But I am Sending Notification From Server Only Once. And Yes  2 notificaitons exactly the same Content.

**Comment by momo1** [comment]
Update:
I coincidentally moved to developing the project on a different machine. It's working perfectly now. I have no idea why that would be the case but I don't see the problem anymore.

I won't close the issue because there are others now still on the thread. If it's just down to me please feel free to close it.

Thanks again

**Comment by kckunal2612** [comment]
Even I am facing this issue and have posted about it in another issue. But I wanted to specify here that I have not registered the service twice like you said, and have only 2 declarations in the Manifest -

```
  <service android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationRegistrationService"/>
        <service
            android:name="com.dieam.reactnativepushnotification.modules.RNPushNotificationListenerService"
            android:exported="false" >
            <intent-filter>
                <action android:name="com.google.android.c2dm.intent.RECEIVE" />
            </intent-filter>
        </service>
```

In spite of this, I am getting duplicate notifications.

**Comment by prasanthyejje** [comment]
@evollu  It's working Fine now. I moved the FCM Code into Another Component.

**Comment by evollu** [comment]
cool

**Comment by Kunalpaul12** [comment]
Please check settings.gradle file
My issues:- I was calling this twice
include ':react-native-fcm'
project(':react-native-fcm').projectDir = new File(rootProject.projectDir, '../node_modules/react-native-fcm/android')

check this ones

**Comment by ivanfranchi** [comment]
I was getting duplicated notifications because of developing 2 apps with the same package name.  
Both the applications were receiving the Notification

**Comment by abhaykumar1415** [comment]
I had to remove
`<receiver android:name="com.google.android.gms.gcm.GcmReceiver" android:exported="true" android:permission="com.google.android.c2dm.permission.SEND" > <intent-filter> <action android:name="com.google.android.c2dm.intent.RECEIVE" /> <category android:name="${applicationId}" /> </intent-filter> </receiver>`

from AndroidManifest.xml

**Comment by EKoetsjarjan** [comment]
com.google.android.gms.gcm.GcmReceiver still got merged from the expo libs,
had also to clean and rebuild project to resolve this issue

**Comment by mohdaamir8182** [comment]
> I had to remove
> `<receiver android:name="com.google.android.gms.gcm.GcmReceiver" android:exported="true" android:permission="com.google.android.c2dm.permission.SEND" > <intent-filter> <action android:name="com.google.android.c2dm.intent.RECEIVE" /> <category android:name="${applicationId}" /> </intent-filter> </receiver>`
> 
> from AndroidManifest.xml

This is a working hack but it creates one issue that is "We cannot show local notification".
